File: leetcode2/0496. Next Greater Element I.py
class Solution(object):
    def nextGreaterElement(self, findNums, nums):
        """
        :type findNums: List[int]
        :type nums: List[int]
        :rtype: List[int]
        """
        dictnums={}
        for k in range(len(nums)):
            j=1
            while k+j<len(nums):
                if nums[k+j]>nums[k]:
                    dictnums[nums[k]]=nums[k+j]
                    break
                else:
                    j+=1
            if k+j==len(nums):
                dictnums[nums[k]]=-1
        rep=[]
        for num in findNums:
            rep.append(dictnums[num])
        return rep

File: archives/C#/0496. Next Greater Element I.cs
public class Solution {
    public int[] NextGreaterElement(int[] findNums, int[] nums) {
        Dictionary<int,int> nextGreaterNumDict= new Dictionary<int,int>();
        for(int i=0;i<nums.Length;i++){
            int flag=-1;
            for(int j=i+1;j<nums.Length;j++){
                if(nums[j]>nums[i]){
                    flag=nums[j];
                    break;
                }
            }
            nextGreaterNumDict[nums[i]]=flag;
        }
        int[] result=new int[findNums.Length];
        for(int i=0;i<findNums.Length;i++){
            result[i]=nextGreaterNumDict[findNums[i]];
        }
        return result;
    }
}
